api/src/main/java/pl/kolendateam/dadcard/race/RaceController.java

```diff
@@ -29,4 +38,18 @@ public ArrayList<RaceBaseDTO> getAll() {
 
         return MaperListRaceToDTO.toRaceBaseDTO(races);
     }
+
+    @GetMapping("{id}/region")
+    public ArrayList<RegionBaseDTO> getRegionsForRace(@PathVariable int id){
+        Optional<Race> raceOpt = this.raceRepository.findById(id);
+
+        if(!raceOpt.isPresent()){
+            throw new ResponseStatusException(
+           HttpStatus.NOT_FOUND, "Race Not Found");
+        }
+
+        List<Region> regions = List.copyOf(raceOpt.get().getAvailableRegions());
```

Comment:
jak tu działa "List.copyOf"?

Reply:
bo AvailableRegion przychodzi jako Set a nie List. List.copyOf to statyczna (wiesz czym są metody statyczne?) w typie List pozwalająca mapować typy listo podobne jak np Set na List

Reply:
"statyczna" to taka co nic wywołuje objektu tylko część klasy, dobrze mówię?